server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/WriteLoadConstraintDeciderTests.java

```diff
@@ -233,16 +233,6 @@ public void testWriteLoadDeciderCanRemain() {
                 testHarness.routingAllocation
             ).type()
         );
-        assertEquals(
-            "A shard without write load should remain on a node with queuing above the threshold",
-            Decision.Type.YES,
-            writeLoadDecider.canRemain(
-                testHarness.clusterState.metadata().getProject().index(indexName),
-                testHarness.shardRoutingNoWriteLoad,
-                testHarness.aboveQueuingThresholdRoutingNode,
-                testHarness.routingAllocation
-            ).type()
-        );
```

Comment:
Do we have a test to show a shard without write load can still trigger `NOT_PREFERRED`?